### ISSUE TITLE
AppLoading shows error message ("Possible Unhandled Promise Rejection") on iOS only

### ISSUE THREAD

**Issue by lucj** [expo/expo#10958]
### Summary of Issue

When using AppLoading component it works fine on Android but I got an error message on iOS 

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

````
Expo CLI 3.27.13 environment info:
    System:
      OS: macOS 10.15.2
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.18.3 - /usr/local/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 6.14.6 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.9.1 - /Users/luc/.rvm/gems/ruby-2.6.5/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 13.4, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6626763
      Xcode: 11.4.1/11E503a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^39.0.0 => 39.0.3
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: 0.63.3 => 0.63.3
      react-native-web: ~0.13.7 => 0.13.14
    npmGlobalPackages:
      expo-cli: 3.27.13
    Expo Workflow: managed
`````

### Steps to Reproduce

In my RootStack, I have a basic function that fetch some items. Once the fetching is done, the "loaded" state is set to true.

````
export function RootStack() {
  ...
  const [loaded, setLoaded] = useState(false);

  const fetchSomeStuff = async () => {
    // Calling api here...
    return true;
  }

  if (loaded === false) return(
    <AppLoading
      startAsync={ fetchSomeStuff }
      onFinish={() => {
        console.log("onFinish");
        setLoaded(true);
      }}
      onError={(err) => { 
        console.error("onError");
        console.log(err);
        setLoaded(true);
      }}  
    />
  )

  return (
      <Root.Navigator mode="modal">
        <Root.Screen name="Main" component={AppStack} options={{ headerShown: false }} />
        <Root.Screen name="Login" component={LoginStack} options={{ headerShown: false }} />
      </Root.Navigator>
  )
}
`````

### Expected Behavior vs Actual Behavior

I expected the same behavior on both iOS and Android but I only got the following error on iOS (both simulator and real device)

````
[Unhandled promise rejection: Error: Native splash screen is already hidden. Call this method before rendering any view.]
- node_modules/react-native/Libraries/BatchedBridge/NativeModules.js:103:50 in promiseMethodWrapper
- node_modules/@unimodules/react-native-adapter/build/NativeModulesProxy.native.js:15:23 in moduleName.methodInfo.name
- node_modules/expo-splash-screen/build/SplashScreen.js:23:7 in preventAutoHideAsync
- node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
- node_modules/regenerator-runtime/runtime.js:293:29 in invoke
- node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
- node_modules/regenerator-runtime/runtime.js:154:27 in invoke
- node_modules/regenerator-runtime/runtime.js:189:16 in PromiseImpl$argument_0
- node_modules/react-native/node_modules/promise/setimmediate/core.js:45:6 in tryCallTwo
- node_modules/react-native/node_modules/promise/setimmediate/core.js:200:22 in doResolve
- node_modules/react-native/node_modules/promise/setimmediate/core.js:66:11 in Promise
- node_modules/regenerator-runtime/runtime.js:188:15 in callInvokeWithMethodAndArg
- node_modules/regenerator-runtime/runtime.js:211:38 in enqueue
- node_modules/regenerator-runtime/runtime.js:238:8 in exports.async
- node_modules/expo-splash-screen/build/SplashScreen.js:23:7 in preventAutoHideAsync
- node_modules/expo/build/launch/AppLoadingNativeWrapper.js:5:8 in AppLoading#constructor
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:7655:17 in constructClassInstance
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:12377:26 in updateClassComponent
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:19181:22 in beginWork$1
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:18085:22 in performUnitOfWork
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:18013:38 in workLoopSync
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:17977:18 in renderRootSync
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:17674:33 in performSyncWorkOnRoot
* [native code]:null in performSyncWorkOnRoot
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:5321:31 in runWithPriority$argument_1
- node_modules/scheduler/cjs/scheduler.development.js:653:23 in unstable_runWithPriority
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:5316:21 in flushSyncCallbackQueueImpl
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:5304:28 in flushSyncCallbackQueue
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:17718:28 in batchedUpdates$1
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:2492:29 in batchedUpdates
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:2638:16 in _receiveRootNodeIDEvent
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:2713:25 in receiveEvent
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:416:4 in __callFunction
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:109:6 in __guard$argument_0
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:364:10 in __guard
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:108:4 in callFunctionReturnFlushedQueue
* [native code]:null in callFunctionReturnFlushedQueue
````

Any hints ?

**Comment by andreas-arkulpa** [comment]
can confirm the same issue

**Comment by hogrid** [comment]
same here

**Comment by nativeforest** [comment]
same here in IOS

**Comment by nativeforest** [comment]
same here in IOS,and the app ge freezed in the screenSplash btw

**Comment by brentvatne** [comment]
post a reproducible example: https://stackoverflow.com/help/minimal-reproducible-example

otherwise not much we can do to help

**Comment by mgiorgi-github** [comment]
Same issue

**Comment by headwinds** [comment]
I'm also experiencing this `[Unhandled promise rejection: Error: Native splash screen is already hidden` and it feels like @lucj might be on to something when he references [AppLoading](https://docs.expo.io/versions/latest/sdk/app-loading/) - there's definitely appears to be relationship between `expo-app-loading` and  `expo-splash-screen`
 
I was using both libraries then I tried removing `expo-splash-screen` but keeping `expo-app-loading` to load assets. I don't have a solution nor a reproducible example yet...  I've upgraded to the latest version 40 too.

**Comment by headwinds** [comment]
I was able to resolve my error but reviewing the docs - thanks for the tip on where to look @lucj 

**tldr;** My error went after following the doc - go figure 😅

In my case, I was using AppLoading in two places - on the splash and on another page. On this other page, I wasn't using all the required props too. I realized I didn't need to do on demand loading and could just do it all in one place on the first splash screen.

**Comment by jer-sen** [comment]
Same issue here on iPhone 5s iOS 12.5.2 and iPhone 7 iOS 14.4.1.
I use `AppLoading` component with this code in the `startAsync` function:

```
		Updates.addListener((event) => {
			if (event.type === Updates.UpdateEventType.UPDATE_AVAILABLE) {
				void Updates.reloadAsync();
			}
		});
```

@brentvatne do you think it could come from the app being reloaded after the splash screen was hidden? Is splash screen redisplayed when the app is reloaded? If I'm right it will be difficult to provide a reproducible example since it can not be reproduced with Expo Go.

**Comment by brentvatne** [comment]
@jer-sen - yeah, i'm pretty sure this is the same issue as https://github.com/expo/expo/issues/11943

**Comment by PeterAbdalla** [comment]
Still an issue
